### PR TITLE
ONNX export: Add Flatten before Gemm

### PR DESCRIPTION
**Pull request by vandanavk**
## Description ##
ONNX's spec specifies that Gemm input data should be 2D, but this wasn't the case for VGG and ResNet models trained in MXNet and exported to ONNX. This PR adds a flatten node before Gemm to make the input data 2D.

Related issues: https://github.com/onnx/models/issues/90, https://github.com/onnx/onnx/issues/1101, https://github.com/onnx/models/issues/91#issuecomment-438782534

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Add a flatten node before Gemm

## Comments ##
- Testing done: 

1. Exported all VGG models (https://github.com/onnx/models/tree/master/models/image_classification/vgg) and ResNetv1 models (https://github.com/onnx/models/tree/master/models/image_classification/resnet) and executed the validation notebook.
2. A test in mxnet_export_test for fully connected

### CODE REVIEW

**Comment by vandanavk** [comment]
@anirudhacharya the test is WIP (added a note in the PR description). Adding a test for fully connected in mxnet_export_test.

**Comment by vandanavk** [comment]
@mxnet-label-bot add [ONNX, pr-work-in-progress]

File: python/mxnet/contrib/onnx/mx2onnx/export_onnx.py [hunk 1]
```diff
@@ -294,7 +298,7 @@ def create_onnx_graph_proto(self, sym, params, in_shape, in_type, verbose=False)
                     # If converted node is NodeProto, add it in processed nodes list
                     elif isinstance(converted_node, NodeProto):
                         onnx_processed_nodes.append(converted_node)
-                        if idx == (len(mx_graph) - 1):
+                        if converted_node.name in output_names:
```
Comment (vandanavk) [inline python/mxnet/contrib/onnx/mx2onnx/export_onnx.py]: This needs modification for cases when there is no converted_node.name - working on this.

**Comment by vandanavk** [comment]
@mxnet-label-bot update [ONNX, pr-awaiting-review]

**Comment by nswamy** [comment]
@Roshrini, @anirudhacharya for review?

**Comment by nswamy** [comment]
@vandanavk 
Please review the CI failure

```
  _, out_shapes, _ = sym.infer_shape(**inputs)

Traceback (most recent call last):

  File "tests/python-pytest/onnx/export/mxnet_export_test.py", line 486, in <module>

    test_models("bvlc_googlenet", (1, 3, 224, 224), (1, 1000))

  File "tests/python-pytest/onnx/export/mxnet_export_test.py", line 142, in test_models

    converted_model_path = onnx_mxnet.export_model(sym, params, [input_shape], np.float32, onnx_file)

  File "/work/mxnet/python/mxnet/contrib/onnx/mx2onnx/export_model.py", line 87, in export_model

    verbose=verbose)

  File "/work/mxnet/python/mxnet/contrib/onnx/mx2onnx/export_onnx.py", line 211, in create_onnx_graph_proto

    graph_outputs = MXNetGraph.get_outputs(sym, params, in_shape, output_label)

  File "/work/mxnet/python/mxnet/contrib/onnx/mx2onnx/export_onnx.py", line 152, in get_outputs

    assert len(out_shapes) == len(out_names)
```

**Comment by vandanavk** [comment]
@zhreshold for review

**Comment by Roshrini** [comment]
@zhreshold Can you please review/merge this PR?

**Comment by vandanavk** [comment]
@mxnet-label-bot update [ONNX, pr-awaiting-merge]

**Pull request merged**